### PR TITLE
Sublime Text support; Windows support; Tracking document changes

### PR DESCRIPTION
**Pull request by alexsaveliev**
This PR is a sidekick of sourcegraph/sublime-lsp#5. It does include:

- Sublime Text editor support:  Ability to plug JS/TS LSP server in stdio mode to Sublime by using `bin/javascript-typescript-sublime` executable
- Adding range info to hover responses. Sublime Text client relies on this information
- Make tests work on Windows by using random ports for communication between test server and client instead of pipes. Removed `tmp` dependency. Tested LSP server with Sublime Text and VSCode on Windows
- added support of `textDocument/did...` notifications (tested with VSCode)

### CODE REVIEW

File: bin/javascript-typescript-sublime [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
Comment (uforic) [inline bin/javascript-typescript-sublime]: Do we need to call this -sublime? Or is it actually -stdio?

File: src/project-manager.ts [hunk 2]
```diff
@@ -544,19 +630,30 @@ export class ProjectConfiguration {
...
+export function processDiagnostics(connection: IConnection, program: ts.Program) {
+    if (!connection) {
+        return;
+    }
+    const diagnostics = ts.getPreEmitDiagnostics(program);
+    if (!diagnostics || !diagnostics.length) {
+        return;
+    }
+    // TODO
+}
```
Comment (uforic) [inline src/project-manager.ts]: add newline

File: src/test/test-utils.ts [hunk 3]
```diff
@@ -217,3 +206,29 @@ export function symbols(params: rt.WorkspaceSymbolParamsWithLimit, expected: vsc
...
+
+export function change(uri: string, text: string) {
+    channel.clientConnection.connection.sendNotification(rt.TextDocumentDidChangeNotification.type, {
+        textDocument: {
+            uri: uri
+        }, contentChanges: [{
+            text: text
+        }]
+    });
+}
```
Comment (uforic) [inline src/test/test-utils.ts]: newline

File: src/project-manager.ts [hunk 4]
```diff
@@ -544,19 +630,30 @@ export class ProjectConfiguration {
...
+
+export function processDiagnostics(connection: IConnection, program: ts.Program) {
+    if (!connection) {
+        return;
+    }
+    const diagnostics = ts.getPreEmitDiagnostics(program);
+    if (!diagnostics || !diagnostics.length) {
+        return;
+    }
+    // TODO
```
Comment (beyang) [inline src/project-manager.ts]: remove or address TODO

**Comment by beyang** [comment]
LGTM after Matt's and my comments are addressed.

**Pull request merged**